engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -241,7 +242,7 @@ public LogChannelInterface getLogChannel() {
     return log;
   }
 
-  public String getXML() {
+  public synchronized String getXML() {
```

Comment:
Yeah, I agree that Reentrant lock would be a better solution.
I'm just saing that this this should be synchronized.